src/composables/queries/usePoolSnapshotsQuery.ts

```diff
@@ -48,18 +56,57 @@ export default function usePoolSnapshotsQuery(
   const queryFn = async () => {
     if (!pool.value) throw new Error('No pool');
 
-    // TODO - remove this once coingecko supports wstEth
-    const tokens = pool.value.tokenAddresses.includes(addresses.wstETH)
-      ? [...pool.value.tokenAddresses, addresses.stETH]
-      : pool.value.tokenAddresses;
+    let tokens: string[] = [];
+
+    const isStablePhantomPool = isStablePhantom(pool.value.poolType);
+
+    if (isStablePhantomPool && pool.value.mainTokens != null) {
+      // TODO - remove this once coingecko supports linear tokens
+      tokens = pool.value.mainTokens;
+    } else if (pool.value.tokenAddresses.includes(addresses.wstETH)) {
+      // TODO - remove this once coingecko supports wstEth
+      tokens = [...pool.value.tokenAddresses, addresses.stETH];
+    } else {
+      tokens = pool.value.tokenAddresses;
+    }
+
+    const [historicalPrices, snapshots] = await Promise.all([
+      coingeckoService.prices.getTokensHistorical(tokens, days),
+      balancerSubgraphService.poolSnapshots.get(id, days)
+    ]);
+
+    if (isStablePhantomPool) {
+      const tokensWithoutPoolToken = tail(pool.value.tokens);
+
+      return {
+        // Adjust the price from coingecko by the priceRate (1 LINEAR-AUSDC -> priceRate * 1 USDC)
+        prices:
+          tokensWithoutPoolToken.length === tokens.length
+            ? mapValues(historicalPrices, prices =>
+                prices.map((price, index) => {
+                  // indexes of the tokens and mainTokens should match
+
+                  const token = tokensWithoutPoolToken[index];
```

Comment:
This is kinda tricky... since what I've sent to coingecko are the `mainTokens` I need to figure out what was the `priceRate` of the `LINEAR-A{TOKEN}`, so for that I need to grab it from the `tokens`.

Another potential solution is perhaps to get it from `onchain.linearPools` - but I thought its kinda the same?

Reply:
The formula is
```
LINEAR-AUSDC * priceRate = USDC
```

Reply:
I think you should get the price rate via the linearPools map, e.g.
```
onchain.linearPools[token.address].priceRate
```